apis/v1alpha1/shared_types.go

```diff
@@ -181,23 +194,25 @@ type RouteGatewayStatus struct {
 	//
 	// +listType=map
 	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=8
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	Conditions []metav1.Condition `json:"conditions"`
 }
 
 // RouteStatus defines the observed state that is required across
 // all route types.
 type RouteStatus struct {
-	// Gateways is a list of the Gateways that are associated with the
-	// route, and the status of the route with respect to each of these
-	// Gateways. When a Gateway selects this route, the controller that
-	// manages the Gateway should add an entry to this list when the
-	// controller first sees the route and should update the entry as
-	// appropriate when the route is modified.
-	//
-	// A maximum of 100 Gateways will be represented in this list. If this list
-	// is full, there may be additional Gateways using this Route that are not
-	// included in the list.
+	// Gateways is a list of Gateways that are associated with the route,
+	// and the status of the route with respect to each Gateway. When a
+	// Gateway selects this route, the controller that manages the Gateway
+	// must add an entry to this list when the controller first sees the
+	// route and should update the entry as appropriate when the route is
+	// modified.
+	//
+	// A minimum of 1 and a maximum of 100 Gateways will be represented
+	// in this list. If the list is full, additional Gateways may be
+	// using this Route that are not included in the list. An empty list
+	// means the route has not been admitted by any Gateway.
```

Comment:
I think this is likely covered better with https://github.com/kubernetes-sigs/service-apis/pull/528, can you pull this change out of this PR?

Reply:
@robscott done.